doc/api/child_process.markdown

```diff
@@ -392,76 +392,202 @@ controlling terminal.
+
+    child.on('close', function(code) {
+      process.exit(code);
+    });
+
+##### `'ipc'`
+
+Creates an [inter-process communication][] (IPC) channel for passing messages
+and/or other file descriptors between the parent process and a child process.
+Only *one* IPC channel may be configured per child. Setting this option enables
+the `ChildProcess#send()` function and the `ChildProcess#on('message')`
+event handler within the parent.
 
-    // Child will use parent's stdios
-    spawn('prg', [], { stdio: 'inherit' });
+If the spawned child is a Node.js process, the presence of an IPC channel
+enables the `process.send()` and `process.disconnect()` functions as well as the
+`process.on('disconnect')` and `process.on('message')` event handlers within the
+child. However, the child's IPC channel is unreferenced until it a
+`process.on('disconnect')` or `process.on('message')` event handler is
+registered. This allows the child to exit normally without being kept alive by a
+referenced IPC channel.
 
-    // Spawn child sharing only stderr
-    spawn('prg', [], { stdio: ['pipe', 'pipe', process.stderr] });
+**NOTE:** The [`child_process.fork()`][] function creates an IPC channel using
+this technique.
 
-    // Open an extra fd=4, to interact with programs presenting a
-    // startd-style interface.
-    spawn('prg', [], { stdio: ['pipe', null, null, null, 'pipe'] });
+The following `parent.js` example spawns a child process with an IPC channel and
+sends it a JSON message. If the IPC channel is referenced by the child process,
+you can type `Ctrl-C` to quit.
+
+    const spawn = require('child_process').spawn;
+    const child = spawn('node', ['child.js'], { stdio: [0, 1, 2, 'ipc'] });
+
+    child.on('message', function(msg) {
+      console.log('PARENT: Received message', msg);
+    });
+
+    child.send({ hello: 'from parent' }, function(err) {
+      if (err) {
+        throw err;
+      }
+      console.log('PARENT: Sent message');
+    });
+
+The following is a complimentary `child.js` example that sends a JSON message to
+its parent through the shared IPC channel.
+
+    process.on('message', function(msg) {
+      console.log('CHILD: Received message', msg);
+    });
+
+    process.send({ hello: 'from child' }, function(err) {
+      if (err) {
+        throw err;
+      }
+      console.log('CHILD: Sent message');
+    });
+
+##### `'pipe'`
+
+Shares a child process's file descriptor with its parent process. The parent can
+access the child's file descriptor using the `ChildProcess#stdio` property. File
+descriptors 0, 1, and 2 are special in that they're also available to the parent
+via the `ChildProcess#stdin`, `ChildProcess#stdout`, and `ChildProcess#stderr`
+properties respectively.
+
+The following example spawns a child process that pipes its stdin, stdout, and
+stderr to its parent. In this example, it might be more effective to share the
+parent's standard streams with the child instead. See the `Integer` file
```

Comment:
you are just saying its a bad example! :-) why not spawn `node -p process.version` and then do something with the output. or pipe a CLI to an http response... or something.